### PR TITLE
Featured profiles

### PR DESCRIPTION
**Pull request by sabrina-ulbrich**
The question is how the translation works in this case. I just get the title and description from database. Should we write the translation also into the database?
screen shot 2018-10-01 at 22 35 09

### CODE REVIEW

**Comment by sabrina-ulbrich** [comment]
When I tried to add a factory for FeaturedProfiles I get the error PG::UndefinedTable...
What do I wrong?
My factory:
`factory :featured_profile do
    title 'gaming'
  end`
Test:
`  let!(:featured_profile) { FactoryBot.create(:featured_profile) }

  describe 'featured profile' do
    it "has a title" do
      expect(featured_profile.title).to eq 'gaming'
    end
  end`

**Comment by sabrina-ulbrich** [comment]
Got it!

`rails db:test:prepare` makes it working

**Comment by tyranja** [comment]
Oh, good question! Totally forgot about the translation. We would need a globalize migration (https://github.com/globalize/globalize#model-translations). The gem ist already installed.

Then we have extra tables for the german and english text.

**Comment by tyranja** [comment]
Cool, you even added them to the admin interface, but how do I get to the page? The link has to be added somewhere here: https://github.com/rubymonsters/speakerinnen_liste/blob/master/app/views/admin/dashboard/index.html.erb I guess.

Do you need to add the featured_profile also to the routes.rb?

How did you get to the view locally?

**Comment by sabrina-ulbrich** [comment]
@tyranja sorry, did this already, but added the code from a wrong directory, so the files were not here, pushed it just now. The array of id's does not work yet, possibility for editing and deleting coming soon, also tests should work again.

**Comment by tyranja** [comment]
Ahh, I can add new featured_profiles in the admin view now :tada: But they are not set `public=true` yet so they are not shown in the view yet.

File: db/migrate/20180924140846_create_featured_profiles.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class CreateFeaturedProfiles < ActiveRecord::Migration[5.2]
+  def change
+    create_table :featured_profiles do |t|
+      t.string :title
+      t.text :description
+      t.string :profile_ids, array: true, default: []
```
Comment (tyranja) [inline db/migrate/20180924140846_create_featured_profiles.rb]: Why are the id's strings and not integers?

File: app/models/featured_profile.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  translates :title, :description
+  globalize_accessors :locales => [:de, :en], :attributes => [:title, :description]
+
+  def self.featured_women
+    featured = FeaturedProfile.find_by(public: true)
+    if featured == nil
+      return nil
+    else
+      featured_profiles = []
+      featured.profile_ids.each { | id | featured_profiles << Profile.find(id) }
```
Comment (tyranja) [inline app/models/featured_profile.rb]: Now when an id is wrong and not in the database you get an error on the startpage. we should make sure that the startpage is still working.

Sth like `if Profile.find(id)` ?

File: db/migrate/20180924140846_create_featured_profiles.rb (hunk 1 above)
Comment (sabrina-ulbrich) [reply db/migrate/20180924140846_create_featured_profiles.rb]: @tyranja do you know how to change the datatype of strings to integer? There is a string_to_array method but no integer_to_array.

Comment (tyranja) [reply db/migrate/20180924140846_create_featured_profiles.rb]: I have found this: https://stackoverflow.com/questions/22394210/rails-migration-to-convert-string-to-integer-using-conversion

Comment (sabrina-ulbrich) [reply db/migrate/20180924140846_create_featured_profiles.rb]: I tried that, but get the following error `PG::CannotCoerce: ERROR:  cannot cast type character varying[] to integer`

Comment (sabrina-ulbrich) [reply db/migrate/20180924140846_create_featured_profiles.rb]: So, in the end I rename the old column (maybe delete it later) and add a new one with an integer array. That is accepted by PG.

File: app/models/featured_profile.rb (hunk 2 above)
Comment (sabrina-ulbrich) [reply app/models/featured_profile.rb]: thanks for suggestion, I add this to the next commit

**Comment by sabrina-ulbrich** [comment]
So, since featured_profiles.title doesn't exist anymore, how do I test translation tables?

File: spec/models/featured_profile_spec.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+describe FeaturedProfile, type: :model do
+  let!(:profile) { FactoryBot.create(:profile, id: 2) }
+
+  describe 'featured profile' do
```
Comment (tyranja) [inline spec/models/featured_profile_spec.rb]: I love that test :grinning:

**Comment by tyranja** [comment]
_First I have merged master to the branch to have the newest changes._

I tried the branch locally and there was just one thing not working ( as far as I noticed ): 
When I am on the `featured profile admin page` and want to edit an already existing `featured profile event`, the id's of the speakerinnen are not longer with comma separated in the edit view. The commas vanished :woman_shrugging:  

And the view of the `öffentlich` button looks strange, maybe you could name it just `public` and `hidden`?
featured_frauen
feat2

Also I was thinking that we might need more space than just 4 featured women and maybe we should take out view of the other speakers? I would like to discuss that with @zaziemo as well and maybe others?

feat3
 
Other than that it is perfect. Just delete the empty files and add some tests and it should be done! Jiphhiiii

**Comment by sabrina-ulbrich** [comment]
Now it is possible to show more than one event. That raises the question if there should be a defined order (another database field for a number of ordering?). Another improvement I think about is to change the implementation to a more rails-way (something like has-many-through-relation), because now every featured profile makes a database query. But thats maybe not necessary for the first release.

**Comment by tyranja** [comment]
I have added an `if` clause in the profile partial because I got an error message locally because one of the id's where wrong and there was no existing profile for that. I image that could happen sometimes and it would be fatal if then the whole page does not work. 

Other than that it is super!

**Pull request merged**